agent/config/parse_linux.go

```diff
@@ -18,10 +18,30 @@ package config
 
 import (
 	"errors"
+	"os"
 	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils"
 )
 
 func parseGMSACapability() bool {
+	envStatus := utils.ParseBool(os.Getenv("ECS_GMSA_SUPPORTED"), true)
+	if envStatus {
+
+		// check if the credentials fetcher socket is created and exists
+		// this env variable is set in ecs-init module
+		if credentialsfetcherHostDir := os.Getenv("CREDENTIALS_FETCHER_HOST_DIR"); credentialsfetcherHostDir != "" {
+			_, err := os.Stat(credentialsfetcherHostDir)
+			if err != nil {
+				if os.IsNotExist(err) {
+					return false
+				}
+			}
+			// returns true if the container instance is domain joined
+			// this env variable is set in ecs-init module
+			return utils.ParseBool(os.Getenv("ECS_DOMAIN_JOINED_LINUX_INSTANCE"), false)
```

Comment:
rather than directly return, I would parse this and log an error if it's not true. If we get to this point, we would expect that this env var would always be set right?

Reply:
added variable before returning